hw/ip/tlul/common.core

```diff
@@ -8,6 +8,7 @@ description: "TL-UL common building blocks"
 filesets:
   files_rtl:
     depend:
+      - lowrisc:dv:pins_if
```

Comment:
@msfschaffner I found we need to add this back for `pinmux_fpv` to run.
Change refers to PR https://github.com/lowRISC/opentitan/pull/1287

Reply:
I see... seems like we should set up an FPV nightly as soon as possible to detect issues like these earlier ;). 

The problem here is that this is DV code which gets included in pure RTL builds. DC doesn't like the default expressions (`int idx = 0`) in the functions inside `pins_if`, and errors out... 

It looks like `pins_if` is required in the TL-UL assertion file. How about we add an `` `ifndef SYNTHESIS`` around the contents of the interface body?